### PR TITLE
Mongodb

### PR DESCRIPTION
**Pull request by t0bbl**

### CODE REVIEW

File: components/Form.js [hunk 1]
```diff
@@ -1,40 +1,109 @@
...
+async function sendRequest(url, { arg: shirtData }) {
+  const response = await fetch(url, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(shirtData),
+  });
+  const { status } = await response.json();
+  console.log(status + " from POST request");
```
Comment (onemanwenttomow) [inline components/Form.js:33]: We don't want console logs making it into the main branch

File: components/Form.js [hunk 2]
```diff
@@ -1,40 +1,109 @@
...
-        required
-      />
-      <Input
-        id="thirdKeyword"
-        name="thirdKeyword"
-        type="text"
-        placeholder="THIRD"
-        required
-      />
+      <Input name="keywordOne" type="text" placeholder="FIRST" required />
```
Comment (onemanwenttomow) [inline components/Form.js:104]: For accessibility reasons we should really have labels for each of these inputs.

File: components/FourPicsTable.js [hunk 3]
```diff
@@ -1,47 +1,70 @@
...
+import { useRouter } from "next/router";
+import useSWR from "swr";
+
+const fetcher = (url) => fetch(url).then((res) => res.json());
 
 export default function FourPicsTable() {
+  const router = useRouter();
+  const { searchID } = router.query;
+
+  const {
```
Comment (onemanwenttomow) [inline components/FourPicsTable.js:13]: At the moment, when the page first renders, `searchID` is `undefined`. That means we would end up making a request to `/api/ChooseVariation/undefined` which i think would return a 404 error. We can prevent that by either providing a url to useSwr or `null`. Something like this. 

``` js 
const {
  data: shirts,
  isLoading,
  error,
} = useSWR(searchID ? `/api/ChooseVariation/${searchID}` : null, fetcher);

``` 

We would then also need to make sure to return loading if there are no shirts yet (i.e on the first render)

``` js
if (isLoading || !shirts) {
  return <div>loading...</div>;
}
```

You can read more on the subject here 
[conditional fetching with useSWR](https://swr.vercel.app/docs/conditional-fetching)

File: components/FourPicsTable.js [hunk 4]
```diff
@@ -1,47 +1,70 @@
...
+    <>
+      {shirts.map((shirt) => (
+        <FourPicsContainer key={shirt._id}>
+          <StyledA href={`/PreviewPage/${shirt.pic1.picSRCSlug}`}>
+            <StyledImage
+              alt="image number 1"
+              src={shirt.pic1.picSRC}
+              width="400"
+              height="400"
+              $fourpictures
```
Comment (onemanwenttomow) [inline components/FourPicsTable.js:35]: We should aim for consistency of prop order. In 3 of the images `fourpictures` is the first prop you pass to the `StyledImage` but in this case it is the last prop. (This is a minor point and won't affect the outcome, just the readability of the code)

File: components/FourVariationsTable.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+const fetcher = (url) => fetch(url).then((res) => res.json());
+
+export default function FourPicsTable() {
+  const router = useRouter();
+  const { searchID } = router.query;
+  const {
+    data: shirts,
+    isLoading,
+    error,
+  } = useSWR(`/api/ChooseVariation/${searchID}`, fetcher);
```
Comment (onemanwenttomow) [inline components/FourVariationsTable.js:16]: Again, we should consider that `searchID` might be undefined on the first render and we should handle that. (see previous comment).

File: db/models/Shirt.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
+import mongoose from "mongoose";
+
+const { Schema } = mongoose;
+
+const shirtSchema = new Schema({
+  searchID: { type: String, required: false },
+  pic1: {
```
Comment (onemanwenttomow) [inline db/models/Shirt.js:7]: There feels like some repetition going on here. i.e. `pic1`, `pic2`, `pic3` and `pic4` all look very similar. 

We could move them to their own schema and then do something like this. 

``` js 
const picSchema = new Schema({
  picSRC: String,
  picSRCSlug: { type: String, required: false },
  picSRCSlugJPG: { type: String, required: false },
  variant1: { type: String, required: false },
  variant1slug: { type: String, required: false },
  variant2: { type: String, required: false },
  variant2slug: { type: String, required: false },
  variant3: { type: String, required: false },
  variant3slug: { type: String, required: false },
  variant4: { type: String, required: false },
  variant4slug: { type: String, required: false },
});

const shirtSchema = new Schema({
  searchID: { type: String, required: false },
  pic1: picSchema,
  pic2: picSchema,
  pic3: picSchema,
  pic4: picSchema,
  keywordOne: { type: String, required: true },
  keywordTwo: { type: String, required: true },
  keywordThree: { type: String, required: true },
});
```

File: pages/PreviewPage/[searchID].js [hunk 7]
```diff
@@ -5,32 +5,55 @@ import Header from "../../components/Header";
...
+  const { searchID } = router.query;
+  const option = router.query.option;
 
-  function handleVariations() {
-    router.push(`/Variations/${imageID}`);
+  const {
+    data: shirts,
+    isLoading,
+    error,
+  } = useSWR(`/api/ChooseVariation/${searchID}`, fetcher);
```
Comment (onemanwenttomow) [inline pages/PreviewPage/[searchID].js:21]: Again, we should consider that `searchID` might be undefined on the first render and we should handle that. (see previous comment).

**Review by onemanwenttomow — CHANGES_REQUESTED** [review]
Overall looks really good. Added a few comments.

File: db/models/Shirt.js (hunk 6 above)
Comment (t0bbl) [reply db/models/Shirt.js:7]: great idea! thank you!

File: components/FourPicsTable.js (hunk 3 above)
Comment (t0bbl) [reply components/FourPicsTable.js:13]: thank you very much, i kind of went crazy over the course of all these dynamic routes

File: components/Form.js [hunk 8]
```diff
@@ -1,38 +1,124 @@
...
 
+    trigger(shirtData);
+    router.push(`/ChooseFour/${searchID}`);
+  }
   return (
     <FormContainer onSubmit={handleSubmit}>
       <Input
-        id="firstKeyword"
-        name="firstKeyword"
+        name="keywordOne"
```
Comment (onemanwenttomow) [inline components/Form.js:104]: Apologies if i wasn't clear before, but for accessibility reasons i think we would want to use the `<label>` element. 

[See here](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label)

**Review by onemanwenttomow — CHANGES_REQUESTED** [review]
LGTM (just one small thing)

File: components/Form.js (hunk 8 above)
Comment (onemanwenttomow) [reply components/Form.js:104]: e.g. something like this

``` js
<label htmlFor="keywordOne">First</label>
<Input
  name="keywordOne"
  id="keywordOne"
  type="text"
  placeholder="FIRST"
  required
/>
```

**Comment by onemanwenttomow** [comment]
LGTM!

**Pull request merged**